File: WebRouterApp/ClientApp/src/app/trading/models/messages/positions-snapshot.message.ts
import { Guid } from '../../../shared/types/guid';
import { IPosition } from '../position';
import { TradingMessageTags } from './trading-message-tags';

export interface IPositionsSnapshotMessage {
  tag: TradingMessageTags.PositionsSnapshot;
  traderId: Guid;
  positions: IPosition[];
}

File: WebRouterApp/Features/Trading/Application/MessageParts/PositionsSnapshotMessage.cs
using System;
using System.Collections.Generic;

namespace WebRouterApp.Features.Trading.Application.MessageParts
{
    public class PositionsSnapshotMessage : ITradingMessage
    {
        public PositionsSnapshotMessage(Guid traderId, IReadOnlyList<PositionModel> positions)
        {
            TraderId = traderId;
            Positions = positions;
        }
        
        public string Tag => "Trading.PositionsSnapshot";

        public Guid TraderId { get; }
        public IReadOnlyList<PositionModel> Positions { get; }
    }
}